src/OpenTelemetry.Instrumentation.Wcf/WcfInstrumentationOptions.cs

```diff
@@ -62,4 +63,28 @@ public class WcfInstrumentationOptions
     /// Gets or sets a value indicating whether or not the SOAP message version should be added as the <see cref="WcfInstrumentationConstants.SoapMessageVersionTag"/> tag. Default value: False.
     /// </summary>
     public bool SetSoapMessageVersion { get; set; }
+
+    /// <summary>
+    /// Gets or sets the reader function which controls exactly how the propagated trace context is extracted from the incoming request.
+    /// Default value: Composed <see cref="TelemetryPropagationReader.HttpRequestHeaders"/> and <see cref="TelemetryPropagationReader.SoapMessageHeaders"/>.
+    /// </summary>
+    /// <remarks>
+    /// <para><see cref="Message"/>: The incoming request (carrier) to read trace context from.</para>
+    /// <para>string: The header name being requested.</para>
+    /// <para>IEnumerable&lt;string&gt; (return): All the values for the requested header, or null if the requested header was not present on the request.</para>
+    /// </remarks>
+    public Func<Message, string, IEnumerable<string>> PropagationReader { get; set; } = TelemetryPropagationReader.Default;
```

Comment:
@repl-chris What is the use case for users needing to mess with these new options? If it isn't super strong and the defaults will work for most users, let's make them `internal`. We can always expose them later if there is a strong need.

Reply:
The main reason I exposed these is because the feature was requested here https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/209 ....but I wouldn't consider that a "strong need".  Theoretically users could be using a transport that the defaults will not work for, but I would expect the defaults to work for the vast majority of users.